spock-core/src/main/java/org/spockframework/util/VersionChecker.java

```diff
@@ -15,15 +15,27 @@
 package org.spockframework.util;
 
 public class VersionChecker {
-  private static final boolean compatibleGroovyVersion =
-      SpockReleaseInfo.isCompatibleGroovyVersion(GroovyReleaseInfo.getVersion());
+  //visibility for testing
+  static final String DISABLE_GROOVY_VERSION_CHECK_PROPERTY_NAME = "spock.iKnowWhatImDoing.disableGroovyVersionCheck";
 
-  public static void checkGroovyVersion(String whoIsChecking) {
-    if (!compatibleGroovyVersion) throw new IncompatibleGroovyVersionException(String.format(
-"The Spock %s cannot execute because Spock %s is not compatible with Groovy %s. For more information, see http://docs.spockframework.org\n" +
+  private static final boolean compatibleGroovyVersion = SpockReleaseInfo.isCompatibleGroovyVersion(GroovyReleaseInfo.getVersion());
+
+  public void checkGroovyVersion(String whoIsChecking) {
+    if (!isCompatibleGroovyVersion() && !isVersionCheckDisabled()) throw new IncompatibleGroovyVersionException(String.format(
```

Comment:
I know that we don't log often but I think that we could log to std.err that the check failed but was ignored due to the property, WDYT?

Reply:
This flag shouldn't be used on a regular basis, so cluttering the stderr to keep users aware sounds fine for me. Changed.

We could check the stderr in tests to verify that message, but 2 ways I know use JUnit 4 rules which we removed from spock-spec ;).